### PR TITLE
fix: throw for missing server routes

### PR DESCRIPTION
**Pull request by caalador**
If serverSideRoutes is missing
from the routes.tsx then
log error on what to do and
throw exception.

Fixes https://github.com/vaadin/flow/issues/18240

### CODE REVIEW

**Review by mshabarov — CHANGES_REQUESTED** [review]
I'm getting the exception even if I have the import in place:

```
com.vaadin.flow.server.ExecutionFailedException: Faulty configuration of serverSideRoutes
        at com.vaadin.flow.server.frontend.TaskGenerateReactFiles.execute(TaskGenerateReactFiles.java:103) ~[flow-server-24.3-SNAPSHOT.jar:24.3-SNAPSHOT]
        at com.vaadin.flow.server.frontend.NodeTasks.execute(NodeTasks.java:342) ~[flow-server-24.3-SNAPSHOT.jar:24.3-SNAPSHOT]
        at com.vaadin.base.devserver.startup.DevModeInitializer.runNodeTasks(DevModeInitializer.java:354) ~[vaadin-dev-server-24.3-SNAPSHOT.jar:na]
```

My routes file:
```JavaScript
import HelloWorldView from 'Frontend/views/helloworld/HelloWorldView.js';
import MainLayout from 'Frontend/views/MainLayout.js';
import { lazy } from 'react';
import { createBrowserRouter, RouteObject } from 'react-router-dom';
import {serverSideRoutes} from "Frontend/generated/flow/Flow";
import {protectRoutes} from "@hilla/react-auth";
import LoginView from "Frontend/views/LoginView";

const AboutView = lazy(async () => import('Frontend/views/about/AboutView.js'));

export const routes: RouteObject[] = protectRoutes([
  {
    element: <MainLayout />,
    handle: { title: 'Main' },
    children: [
      { path: '/', element: <HelloWorldView />, handle: { title: 'Hello World', rolesAllowed: ['USER'] } },
      { path: '/about', element: <AboutView />, handle: { title: 'About' } },
      ...serverSideRoutes
    ],
  },
  { path: '/login', element: <LoginView />},
]);

export default createBrowserRouter(routes);
```

**Comment by mshabarov** [comment]
Also, the helper message isn't really visible, because the application keeps throwing `com.vaadin.flow.server.ExecutionFailedException: Faulty configuration of serverSideRoutes` all the time and what is logged at the beginning was lost in the logs.

**Comment by caalador** [comment]
Changed to use find instead of match.
Put full message into exception.

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateReactFiles.java [hunk 1]
```diff
@@ -54,6 +60,13 @@
 public class TaskGenerateReactFiles implements FallibleCommand {
 
     private final File frontendDirectory;
+    protected static String NO_IMPORT = """
+            Faulty configuration of serverSideRoutes.
+            The server route definition is missing from the '%s' file
+
+            To have working Flow routes add the import 'import "{serverSideRoutes} from \\"Frontend/generated/flow/Flow\\";'
+            and add the route '...serverSideRoutes' into the routes definition.
+            """;
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateReactFiles.java:69]: Let's exclude `"` from the suggested import string:

```
    protected static String NO_IMPORT = """
            Faulty configuration of serverSideRoutes.
            The server route definition is missing from the '%1$s' file.

            To have working Flow routes add the following to the '%1$s' file:
- import line 'import {serverSideRoutes} from 'Frontend/generated/flow/Flow';'
- route '...serverSideRoutes' into the routes definition, example:

export const routes = [
  {
    element: <MainLayout />,
    handle: { title: 'Main' },
    children: [
      { path: '/', element: <HelloWorldView />, handle: { title: 'Hello World' } },
      ...serverSideRoutes
    ],
  },
] as RouteObject[];

            """;
```

**Review by mshabarov — CHANGES_REQUESTED** [review]
Works fine for me.

Have one comment about error message.

As a suggestion: would it be better to insert the import string automatically if it's not there and search for `...serverSideRoutes` entry in the route configuration and throw if not added ?

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateReactFiles.java (hunk 1 above)
Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateReactFiles.java:69]: Currently the log message give me string `import "{serverSideRoutes} from \"Frontend/generated/flow/Flow\";`, which produces errors if I copy-paste it.

Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateReactFiles.java:69]: That was left there by mistake as it was earlier on multiple lines. Missed it on refactor

**Review by mshabarov — APPROVED** [review]

**Pull request merged**